src/plugins/vis_builder/public/application/components/data_tab/use/use_dropbox.tsx

```diff
@@ -156,43 +174,25 @@ export const useDropbox = (props: UseDropboxProps): DropboxProps => {
   );
 
   useEffect(() => {
-    const getValidAggTypes = () => {
-      if (!dragData || schema.group === 'none') return [];
-      const isCountField = dragData === COUNT_FIELD;
-
-      const indexField = isCountField
-        ? { type: 'count' }
-        : getIndexPatternField(dragData, indexPattern?.fields ?? []);
-
-      if (!indexField) return [];
-
-      // Get all aggTypes allowed by the schema and get a list of all the aggTypes that the dragged index field can use
-      const aggTypes = aggService.types.getAll();
-      // `types` can be either a Bucket or Metric aggType, but both types have the name property.
-      const allowedAggTypes = filterByName(
-        aggTypes[schema.group] as BucketAggType[],
-        schema.aggFilter
-      );
-
-      return (
-        allowedAggTypes
-          .filter((aggType) => {
-            const allowedFieldTypes = aggType.paramByName('field')?.filterFieldTypes;
-            return filterByType([indexField], allowedFieldTypes).length !== 0;
-          })
-          .filter((aggType) => (isCountField ? true : aggType.name !== 'count'))
-          // `types` can be either a Bucket or Metric aggType, but both types have the name property.
-          .map((aggType) => (aggType as BucketAggType).name)
-      );
-    };
-
-    setValidAggTypes(getValidAggTypes());
+    const fieldName = typeof dragData === typeof COUNT_FIELD ? '' : (dragData as string);
+    const sourceGroup = typeof dragData === typeof COUNT_FIELD ? 'preDefinedCountMetric' : '';
+
+    setValidAggTypes(
+      getValidAggTypes({
+        fieldName,
+        sourceGroup,
+        destinationSchema: schema,
+        aggProps,
+        aggService,
+        sourceAgg: null,
+      })
+    );
 
     return () => {
       setValidAggTypes([]);
     };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why do we need this?

nit: its always good to mention in a comment why you disable es-lint rules. Its fine to do so, just make sure to leave the justification in the comments

Reply:
Added comments